src/actions/Deselect.ts

```diff
@@ -0,0 +1,41 @@
+import { Selection } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  TypedSelection,
+  Graph,
+} from "../typings/Types";
+import { runOnTargetsForEachEditor } from "../util/targetUtils";
+
+export default class Deselect implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await runOnTargetsForEachEditor(targets, async (editor, targets) => {
+      // Remove selections with a non-empty intersection
+      const newSelections = editor.selections.filter(
+        (selection) =>
+          !targets.find((target) => {
```

Comment:
Total nit here sorry, but

```suggestion
          !targets.some((target) => {
```